lib/block-supports/behaviors.php

```diff
@@ -87,10 +87,18 @@ function gutenberg_render_behaviors_support_lightbox( $block_content, $block ) {
 	// We want to store the src in the context so we can set it dynamically when the lightbox is opened.
 	$z = new WP_HTML_Tag_Processor( $content );
 	$z->next_tag( 'img' );
+
 	if ( isset( $block['attrs']['id'] ) ) {
-		$img_src = wp_get_attachment_url( $block['attrs']['id'] );
+		$img_src      = wp_get_attachment_url( $block['attrs']['id'] );
+		$img_metadata = wp_get_attachment_metadata( $block['attrs']['id'] );
+		$img_width    = $img_metadata['width'];
+		$img_height   = $img_metadata['height'];
+		$img_srcset   = wp_get_attachment_image_srcset( $block['attrs']['id'] );
 	} else {
-		$img_src = $z->get_attribute( 'src' );
+		$img_src        = $z->get_attribute( 'src' );
+		$img_dimensions = getimagesize( $img_src );
```

Comment:
@c4rl0sbr4v0 A note related to performance: I realize this line will fetch an entire image to determine dimensions when an image is inserted via link. We can likely find this information without needing to download the whole image, but that seems beyond the scope of this PR and an enhancement.

What do you think?

Reply:
As you mention, could be out of the scope. But is still an important enhancement. As, for example, in an online newspaper, where a post can contain several images, we can cause unexpected performance problems or increase transfer costs to hosting providers.

Reply:
Done 👍